Parser/Node.cs

```diff
@@ -44,10 +60,46 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
 
+// A Read statement
+public record NReadStmt (Token[] Idents) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
 // An assignment statement
 public record NAssignStmt (Token Name, NExpr Expr) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+// A call statement
+public record NCallStmt (NFnCall FnCall) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A While statement
+public record NWhileStmt (NExpr Exp, NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// An if statement
+public record NIfStmt (NExpr Exp, NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// An else statement
+public record NElseStmt (NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}

```

Comment:
This should be:
`public record NIfStmt (NExpr Condition, NStmt IfPart, NStmt? ElsePart)`